homeassistant/components/google_assistant/smart_home.py

```diff
@@ -168,6 +169,11 @@ def query_serialize(self):
         for trt in self.traits():
             if trt.can_execute(command, params):
                 await trt.execute(command, params)
+                self.hass.bus.async_fire(EVENT_COMMAND_RECEIVED, {
```

Comment:
We should put this before execution. The goal is that the event and the services called will all share the same context, so we can relate one to another.

Reply:
The event is now fired before the execution.
I still have it dis-aggregated however, so there's one event for each device and command. It retains the request_id in the event so they can be re-aggregated if needed.